From a24b99d6679a8ca39fdaa90e35a4fd4c4e3d60c8 Mon Sep 17 00:00:00 2001
From: Martchus <martchus@gmx.net>
Date: Tue, 7 Feb 2017 18:25:28 +0100
Subject: [PATCH 30/31] Prevent qmake from messing static lib dependencies

In particular, it messes resolving cyclic dependency between
static freetype2 and harfbuzz
---
 qmake/generators/unix/unixmake.cpp     | 3 +++
 qmake/generators/win32/winmakefile.cpp | 3 +++
 2 files changed, 6 insertions(+)

diff --git a/qtbase/qmake/generators/unix/unixmake.cpp b/qtbase/qmake/generators/unix/unixmake.cpp
index 30f99174f8..154edd67d4 100644
--- a/qtbase/qmake/generators/unix/unixmake.cpp
+++ b/qtbase/qmake/generators/unix/unixmake.cpp
@@ -414,6 +414,9 @@ UnixMakefileGenerator::fixLibFlag(const ProString &lib)
 bool
 UnixMakefileGenerator::findLibraries(bool linkPrl, bool mergeLflags)
 {
+    if(project->isActiveConfig("staticlib")) {
+        return false; // prevent qmake from messing static lib dependencies
+    }
     QList<QMakeLocalFileName> libdirs, frameworkdirs;
     int libidx = 0, fwidx = 0;
     for (const ProString &dlib : project->values("QMAKE_DEFAULT_LIBDIRS"))
diff --git a/qtbase/qmake/generators/win32/winmakefile.cpp b/qtbase/qmake/generators/win32/winmakefile.cpp
index 2e6d5d94a9..a8320bae09 100644
--- a/qtbase/qmake/generators/win32/winmakefile.cpp
+++ b/qtbase/qmake/generators/win32/winmakefile.cpp
@@ -87,6 +87,9 @@ Win32MakefileGenerator::findLibraries(bool linkPrl, bool mergeLflags)
     impexts.append(project->values("QMAKE_EXTENSION_IMPORTLIB"));
     impexts.append(project->values("QMAKE_EXTENSION_STATICLIB"));
   }
+  if(project->isActiveConfig("staticlib")) {
+    return false; // prevent qmake from messing static lib dependencies
+  }
   QList<QMakeLocalFileName> dirs;
   static const char * const lflags[] = { "QMAKE_LIBS", "QMAKE_LIBS_PRIVATE", 0 };
   for (int i = 0; lflags[i]; i++) {
-- 
2.13.2

